migration/1.3.x-1.4.x/README.md

```diff
@@ -0,0 +1,94 @@
+## Intro
+
+Pachyderm 1.4 has a different data model from the 1.3 series of releases
+and as such it's not possible to map 1.3 data and pipelines 1-to-1 with
+equivalents in 1.4. The biggest change between 1.3 and 1.4 concerns large
+files, in 1.3 it was possible to store very large files and parallelize
+directly over them. While this was convenient for a number of cases it
+wound up iteracting in confusing ways with the filesystem layouts we use
+for data processing and with our incrementality features. It also felt
+a little too magical because it abstracted key details of how computations
+would be parallelized away from the user. In 1.4 files can still be large
+but large files can only be processed by a single process. If data needs
+to be processed in parallel it must be spit into multiple smaller files,
+to help with this syntax has been added to the API to help split files.
+
```

Comment:
Very nice explanation of the reasoning behind making files as the smallest processing units.